### PR TITLE
Switch over local build to use asciidoctor

### PR DESCRIPTION
**Pull request by dedemorton**
Merge this change after the build is switched over to asciidoctor in https://github.com/elastic/docs/pull/1178.

### CODE REVIEW

File: script/build_docs.sh [hunk 1]
```diff
@@ -38,5 +38,5 @@ do
     params="$params --resource=${resource_dir}"
   fi
 
-  $docs_dir/build_docs $params --doc "$index" --out "$dest_dir"
+  $docs_dir/build_docs --asciidoctor $params --doc "$index" --out "$dest_dir"
```
Comment (nik9000) [inline script/build_docs.sh]: I think you want the `--respect_edit_url_overrides` here too.

**Comment by nik9000** [comment]
Lgtm

**Comment by dedemorton** [comment]
@nik9000 How far back do you think I should backport this? I think devs will still want to use the `make docs` and `make docs-preview` commands, but I think that's less likely in older branches. Do you think 7.4 and 7.3 are far back enough?

**Comment by nik9000** [comment]
I'd probably go all the way back just for maximum paranoia's sake. So long as the merge is clean. Once it starts to get unclean I'd think pretty hard about stopping.

**Comment by dedemorton** [comment]
@urso What do think? After spending countless hours backporting the asciidoctor fixes, I don't want to waste time backporting to branches we'll never touch again.

**Pull request merged**